analysis/study_definition.py

```diff
@@ -1,13 +1,884 @@
+      },
+  )
+def comorbidity_icd(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_diagnoses = dx_codelist,
+      returning = "binary_flag",
+      on_or_before = "covid_test_positive_date",
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.1,
+      },
+  )
+def comorbidity_icd_12m(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_diagnoses = dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 12 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.1,
+      },
+  )
+def comorbidity_ops(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_procedures=dx_codelist,
+      returning = "binary_flag",
+      on_or_before = "covid_test_positive_date",
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+          "rate": "exponential_increase",
+      },
+  )
+def comorbidity_ops_12m(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_procedures=dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 12 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.01,
+          "rate": "exponential_increase",
+      },
+  )
+def drug_6m(dx_codelist):
+  return patients.with_these_medications(
+      dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 6 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  )
+def drug_12m(dx_codelist):
+  return patients.with_these_medications(
+      dx_codelist,
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 12 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  )
+def high_cost_drugs_3m(dx_codelist):
+  return patients.with_high_cost_drugs(
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 3 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  ) 
+def high_cost_drugs_6m(dx_codelist):
+  return patients.with_high_cost_drugs(
+      returning = "binary_flag",
+      between = ["covid_test_positive_date - 6 months", "covid_test_positive_date"],
+      find_last_match_in_period=True,
+      return_expectations={
+          "incidence": 0.01,
+      },
+  ) 
+def covid_therapeutics(dx_codelist):
+  return patients.with_covid_therapeutics(
+      dx_codelist,
+      with_these_indications = "non_hospitalised",
+      between = ["covid_test_positive_date", "covid_test_positive_date + 5 days"],
```

Comment:
Does this mean that i will include patients who have had therapeutics outside of the prescribed window of within 5 days of a covid test (if there are any)? or patients given these treatments for other indications?

Reply:
As far as I am aware it is very unlikely for a patient to have been prescribed a therapeutic outside of "the window". And we can't say for sure when this is happening because the 5 day window (at least initially, not sure about more recently) refers to onset of symptoms not test date. We just tend to use test date as a proxy to calculate the window because symptoms are not as well recorded. So in the instances when we see people being given treatment outside of the window this is likely because of how the data is captured rather than anything else. 

Given how this data set has been collected I don't think there is anyway that a patient will have been given these treatments for other indications. 

I believe the numbers are super small anyway so you could include or exclude and I think you'd be justifying the reason either way.

Reply:
thank you, have changed this to index date